src/mlpack/methods/decision_tree/decision_tree.hpp

```diff
@@ -150,6 +163,17 @@ class DecisionTree :
                    typename std::remove_reference<WeightsType>::type>::value>*
                     = 0);
 
+  template<typename MatType, typename LabelsType, typename WeightsType>
+  DecisionTree(const DecisionTree& other,
+               MatType data,
+               LabelsType labels,
+               const size_t numClasses,
+               WeightsType weights,
+               const size_t minimumLeafSize = 10,
+               const double minimumGainSplit = 1e-7,
```

Comment:
If we are using `other` to take parameters, wouldn't `minimumLeafSize` and `minimumGainSplit` be unnecessary?

Reply:
So if we do not pass the these two parameters,  I think we need to add new private member variables to save `minimumLeafSize` and `minimumGainSplit`.